workflows/sarek.nf

```diff
@@ -364,25 +350,35 @@ workflow SAREK {
             // convert back to fastq for further preprocessing
             ALIGNMENT_TO_FASTQ_UMI(CREATE_UMI_CONSENSUS.out.consensusbam, [])
 
-            ch_input_sample_to_split = ALIGNMENT_TO_FASTQ_UMI.out.reads
+            ch_reads_fastp = ALIGNMENT_TO_FASTQ_UMI.out.reads
 
             // Gather used softwares versions
             ch_versions = ch_versions.mix(ALIGNMENT_TO_FASTQ_UMI.out.versions)
             ch_versions = ch_versions.mix(CREATE_UMI_CONSENSUS.out.versions)
         } else {
-            ch_input_sample_to_split = ch_reads
+            ch_reads_fastp = ch_input_fastq
         }
 
-        // SPLIT OF FASTQ FILES WITH SEQKIT_SPLIT2
-        if (params.split_fastq > 1) {
-            SPLIT_FASTQ(ch_input_sample_to_split)
+        // Trimming and/or splitting
+        if (params.trim_fastq || params.split_fastq > 0) {
+            FASTP(ch_reads_fastp, false, false)
```

Comment:
How do you specify here if trimming or not? As I read it here it's always trimming reads, even if `trim_fastq` is false, provided that `split_fastq>0`. Is that intended?

Reply:
In the modules.config it sets:

```ext.args = [ "",
            params.trim_fastq              ?: "--disable_adapter_trimming",
            params.clip_r1 > 0             ? "--trim_front1 ${params.clip_r1}" : "", // Remove bp from the 5' end of read 1.
            params.three_prime_clip_r1 > 0 ? "--trim_tail1 ${params.three_prime_clip_r1}" : "", // Remove bp from the 3' end of read 1 AFTER adapter/quality trimming has been performed.
            params.clip_r2   > 0           ? "--trim_front2 ${params.clip_r2}" : "", //Remove bp from the 5' end of read 2.
            params.three_prime_clip_r2 > 0 ? "--trim_tail2 ${params.three_prime_clip_r2}" : "", // Remove bp from the 3' end of read 2 AFTER adapter/quality trimming has been performed.
            params.trim_nextseq            ? "--trim_poly_g" : "", // Apply the --nextseq=X option, to trim based on quality after removing poly-G tails.
            params.split_fastq > 0         ? "--split_by_lines ${params.split_fastq * 4}" : "",

        ].join(" ").trim()```
        
        so if not params.trim_fastq, aka trimming false, then the tool sets `--disable_adapter_trimming` and shouldn't trim any adapters